docs/tutorials/services/constrain-motion.md

```diff
@@ -0,0 +1,410 @@
+{{< /alert >}}
+
+Before starting this tutorial, make sure you have the [Viam Python SDK](https://python.viam.dev/) <!-- or the [Viam Go SDK](https://pkg.go.dev/go.viam.com/rdk/robot/client#section-readme/) -->installed.
+
+If you are connecting to a real robotic arm during this tutorial, make sure your computer can communicate with the arm controller before continuing.
+
+Familiarize yourself with the concepts outlined in the second motion tutorial, [Plan Motion with an Arm and a Gripper](../plan-motion-with-arm-gripper/), before continuing.
+This tutorial picks up where **Plan Motion with an Arm and a Gripper** stops, so further examples depend on having a connected robot, client and service access, and other infrastructure in place.
+
+For reference, look at [the full code sample from the prior tutorial](../plan-motion-with-arm-gripper/#full-tutorial-code).
+
+## Configure your robot
+
+Use the robot configuration from [the previous tutorial](../plan-motion-with-arm-gripper/) for this tutorial, including the [arm](../../../components/arm/) and [gripper](../../../components/gripper/) components with [frames](../../../services/frame-system/) configured.
+
+The Motion Service is one of the "built-in" services, so you don't need to do anything to enable it on your robot.
+
+{{% expand "Click for an example raw JSON config." %}}
+
+If you go to your robot's **Config** tab on [the Viam app](https://app.viam.com/) and click **Raw JSON**, your config should resemble the following:
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "components": [
+    {
+      "name": "myArm",
+      "type": "arm",
+      "model": "xArm6",
+      "attributes": {
+        "host": "<ip-address>",
+        "port": 0,
+        "speed_degs_per_sec": 15,
+        "acceleration_degs_per_sec_per_sec": 0
+      },
+      "depends_on": [],
+      "frame": {
+        "parent": "world",
+        "translation": {
+          "x": 0,
+          "y": 0,
+          "z": 0
+        },
+        "orientation": {
+          "type": "ov_degrees",
+          "value": {
+            "x": 0,
+            "y": 0,
+            "z": 1,
+            "th": 0
+          }
+        }
+      }
+    },
+    {
+      "name": "myGripper",
+      "type": "gripper",
+      "model": "fake",
+      "attributes": {},
+      "depends_on": [
+        "myArm"
+      ],
+      "frame": {
+        "parent": "myArm",
+        "translation": {
+          "x": 0,
+          "y": 0,
+          "z": 90
+        },
+        "orientation": {
+          "type": "ov_degrees",
+          "value": {
+            "x": 0,
+            "y": 0,
+            "z": 1,
+            "th": 0
+          }
+        },
+        "geometry": {
+          "x": 73,
+          "y": 73,
+          "z": 90,
+          "translation": {
+            "x": 0,
+            "y": 0,
+            "z": 0
+          }
+        }
+      }
+    }
+  ]
+}
+```
+
+{{% /expand %}}
+
+<!--
```

Comment:
what's happening with this comment?

Reply:
I heard that the visualizer stuff would significantly improve soon, and it isn't currently in a showcase-ready state.